notebook/static/notebook/js/codecell.js

```diff
@@ -439,6 +439,29 @@ define([
         this.output_area.toggle_scroll();
     };
 
+    /**
+     * Whether or not the cells input is hidden.
+     * @return {boolean}
+     */
+    Object.defineProperty(CodeCell.prototype, 'hide_input', {
+        get: function() {
+            return this.element.hasClass('hide-input');
+        },
+        set: function(value) {
+            var isHidden = this.element.hasClass('hide-input');
```

Comment:
Presumably changes should be persisted to `this.metadata.hide_input`, and initial state loaded in fromJSON, yes?